src/Enzo/enzo_EnzoMethodBondiHoyleAccretion.cpp

```diff
@@ -0,0 +1,114 @@
+/// See LICENSE_CELLO file for license and copyright information
+
+/// @file	enzo_EnzoMethodBondiHoyleAccretion.cpp
+/// @author     Stefan Arridge (stefan.arridge@gmail.com)
+/// @author     John Regan (john.regan@mu.ie)
+/// @date       13 April 2022
+/// @brief      Computes accretion rates according to Bondi-Hoyle model.
+///             See Krumholz+ 2004, ApJ, 611, 399 for details.
+///
+
+#include "cello.hpp"
+#include "enzo.hpp"
+
+//------------------------------------------------------------------
+
+EnzoMethodBondiHoyleAccretion::EnzoMethodBondiHoyleAccretion
+(double accretion_radius_cells,
+ double density_threshold,
+ double max_mass_fraction)
+  : EnzoMethodAccretion(accretion_radius_cells,
+			density_threshold,
+			max_mass_fraction)
+{
+
+}
+
+//-------------------------------------------------------------------
+
+void EnzoMethodBondiHoyleAccretion::pup (PUP::er &p)
+{
+  // NOTE: Change this function whenever attributes change
+
+  TRACEPUP;
+
+  EnzoMethodAccretion::pup(p); // call parent class pup
+
+  return;
+}
+
+//--------------------------------------------------------------------
+
+void EnzoMethodBondiHoyleAccretion::compute (Block * block) throw()
+{
+
+  if (enzo::simulation()->cycle() == enzo::config()->initial_cycle)
+    do_checks_(block);
+
+  if (block->is_leaf()) {
+    this->compute_(block);
+  }
+  else block->compute_done();
+
+  return;
+}
+
+//-------------------------------------------------------------------------
+
+void EnzoMethodBondiHoyleAccretion::compute_(Block * block)
+{
+  // Only need to do anything if there are sink particles on this block.
+  Particle particle = block->data()->particle();
+  int it = particle.type_index("sink");
+  int num_particles = particle.num_particles(it);
+
+  if (num_particles > 0) {
+
+    // Get pointer to density field data
+    Field field = block->data()->field();
+    enzo_float * density = (enzo_float*) field.values("density");
+
+    // Set accretion radius to be accretion_radius_cells_ multipled by minimum cell width
+    double hx, hy, hz;
+    block->cell_width(&hx, &hy, &hz);
+    const double min_cell_width = std::min(hx,std::min(hy,hz));
+    const double accretion_radius = accretion_radius_cells_ * min_cell_width;
+
+    // Get gravitational constant in code units
+    const double const_G =
+      cello::grav_constant * enzo::units()->mass() * enzo::units()->time() * enzo::units()->time()
+      / (enzo::units()->length() * enzo::units()->length() * enzo::units()->length());
+
```

Comment:
// is it ok to use mass()?
// in enzo_EnzoMethodBackgroundAcceleration.cpp the Gravitational constant is defined as
// double G_code = cello::grav_constant * enzo_units->density() * enzo_units->time() * enzo_units->time();

Reply:
There is no `enzo::units()->density()` in the main branch at the moment: this is something that is in #174 however. If that gets merged into `main` before this does then maybe I can change this.

Reply:
Done.